### PR TITLE
Pass custom properties to persistence backend

### PR DESCRIPTION
**Pull request by tjake**

### CODE REVIEW

File: restapi/src/main/java/io/stargate/web/resources/Db.java [hunk 1]
```diff
@@ -100,7 +103,11 @@ private DataStore getDataStoreInternal(
             .build();
 
     DataStoreOptions options =
-        DataStoreOptions.builder().defaultParameters(parameters).alwaysPrepareQueries(true).build();
+        DataStoreOptions.builder()
+            .defaultParameters(parameters)
+            .alwaysPrepareQueries(true)
+            .customProperties(authenticationSubject.customProperties())
```
Comment (dimas-b) [inline restapi/src/main/java/io/stargate/web/resources/Db.java:109]: This looks a bit inconsistent with the other places in restapi. In other cases we use full HTTP headers, but here only the subject's headers, which is not guaranteed to be the same. One level up the call chain to this method full headers are available... perhaps we could just pass them here as one extra parameter?

As for `AuthenticationSubject`, would it make sense to _combine_ subject properties (may be set by auth services) with the raw headers? The subject's properties are different from raw HTTP headers, but will reflect the authenticators interpretation of the request, which might be worth examining by persistence implementations. WDYT?

Comment (tjake) [reply restapi/src/main/java/io/stargate/web/resources/Db.java:109]: Ideally these would be the raw http headers, I didn't see where those were.

Comment (tjake) [reply restapi/src/main/java/io/stargate/web/resources/Db.java:109]: Would you mind pushing that change?

Comment (dimas-b) [reply restapi/src/main/java/io/stargate/web/resources/Db.java:109]: sure

Comment (dimas-b) [reply restapi/src/main/java/io/stargate/web/resources/Db.java]: implemented that in c16c71458ee7989a06e0222898978775f6dac415

**Review by dimas-b — APPROVED** [review]

File: restapi/src/main/java/io/stargate/web/resources/Db.java [hunk 2]
```diff
@@ -88,19 +92,28 @@ public AuthenticatedDB getDataStoreForToken(
...
         authenticationService.validateToken(token, headers);
     return new AuthenticatedDB(
-        getDataStoreInternal(authenticationSubject, pageSize, pagingState), authenticationSubject);
+        getDataStoreInternal(authenticationSubject, pageSize, pagingState, headers),
+        authenticationSubject);
   }
 
   private DataStore getDataStoreInternal(
-      AuthenticationSubject authenticationSubject, int pageSize, ByteBuffer pagingState) {
+      AuthenticationSubject authenticationSubject,
```
Comment (dougwettlaufer) [inline restapi/src/main/java/io/stargate/web/resources/Db.java:100]: There won't be an authenticationSubject for internal. An "internal" datastore is for things like validating a table based token.

Comment (dougwettlaufer) [reply restapi/src/main/java/io/stargate/web/resources/Db.java:100]: Never mind, was thinking of `createInternal`.

**Review by dougwettlaufer — APPROVED** [review]

**Review by tomekl007 — APPROVED** [review]

**Review by mpenick — APPROVED** [review]

**Pull request merged**